miner/worker.go

```diff
@@ -834,15 +834,16 @@ func (w *worker) commitNewWork(interrupt *int32, noempty bool, timestamp int64)
 	tstart := time.Now()
 	parent := w.chain.CurrentBlock()
 
-	if parent.Time() >= uint64(timestamp) {
-		timestamp = int64(parent.Time() + 1)
-	}
+	// TODO: Is there some other sensible check that we can do in place of the below code with timestamps set from L1?
```

Comment:
Assuming this is the timestamp that will be used for the next sequencer batch, it has to be set far back enough in the past so that there cannot be a new/unseen L1 block with a older timestamp that contains a (Safety or l1toL2) batch. Given that 2 block reorgs are frequent on Ethereum, I think setting timestamp = 2 block old L1 timestamp might be approx what we need.

Reply:
I think that should be handled by the Sync Service. It subscribes to L1 blocks and gets the timestamp from there. Importantly, it includes waiting for a specific number of confirmations before handling an L1 block, so we'll never see a tx from L1 that isn't `min # confirms` old.